ui/apps/platform/src/Containers/Collections/RuleSelector/RuleSelector.cy.jsx

```diff
@@ -0,0 +1,253 @@
+import React, { useEffect, useState } from 'react';
+import isEqual from 'lodash/isEqual';
+
+import RuleSelector from './RuleSelector';
+
+// Component wrapper to allow a higher level component to feed updated state back to the RuleSelector.
+function DeploymentRuleSelector({ defaultSelector, onChange }) {
+    const [resourceSelector, setResourceSelector] = useState(defaultSelector);
+
+    return (
+        <RuleSelector
+            entityType="Deployment"
+            scopedResourceSelector={resourceSelector}
+            handleChange={(_, newSelector) => {
+                setResourceSelector(newSelector);
+                onChange(newSelector);
+            }}
+            validationErrors={undefined}
+        />
+    );
+}
+
+function setup(defaultSelector, onChange) {
+    cy.intercept('GET', '/v1/collections/*/autocomplete', (req) => req.reply([]));
+
+    cy.mount(<DeploymentRuleSelector defaultSelector={defaultSelector} onChange={onChange} />);
+}
+
+describe(Cypress.spec.relative, () => {
+    it('Should render "All entities" option when selector is null', () => {
+        setup({ type: 'All' }, () => {});
+
+        cy.findByText('All deployments');
+    });
+
+    it('Should allow users to add name selectors', () => {
+        let state = {
```

Comment:
Does declaration of `state` need `let` instead of `const`?

Reply:
This should be caught be eslint - I'll have to check and see why these files aren't being picked up.